src/main/java/teammates/common/datatransfer/questions/FeedbackRankRecipientsQuestionDetails.java

```diff
@@ -278,6 +282,23 @@ public String getQuestionResultStatisticsCsv(
     }
 
     /**
+     * Generates a key, value mapping. Each key corresponds to a recipient and its value is the normalised self rank.
+     * @param responses  a list of responses
+     */
+    private Map<String, Integer> generateSelfRankForEachRecipient(List<FeedbackResponseAttributes> responses) {
+        Map<FeedbackResponseAttributes, Integer> normalisedRankOfResponse = getNormalisedRankForEachResponse(responses);
```

Comment:
Currently through this line and line 179, the same step of `getNormalisedRankForEachResponse` on all responses is repeated twice in `getQuestionResultStatisticsHtml`. It would be good if you could find a way to avoid doing this step twice without compromising the code quality. It is not compulsory for you to do this though.